src/include/sof/audio/module_adapter/intel/system_service.h

```diff
@@ -0,0 +1,277 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ */
+/*! \file system_service.h */
+
+
+#ifndef _ADSP_SYSTEM_SERVICE_H_
+#define _ADSP_SYSTEM_SERVICE_H_
+
+#include "core/kernel/logger/log.h"
+#include "adsp_stddef.h"
+#include "adsp_error_code.h"
+
+#include <stdint.h>
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+struct AdspLogHandle;
+typedef struct AdspLogHandle AdspLogHandle;
+
+/*! \brief This struct defines the obfuscating type for notifications. */
+typedef struct _AdspNotificationHandle {} *AdspNotificationHandle;
+
+/*! \brief Defines parameters used by ADSP system during notification creation. */
+typedef struct _NotificationParams {
+	uint32_t type;              /*!< Notification type */
+	uint16_t user_val_1;        /*!< 16 bits user value available directly in IPC header
+				     * for some notifications
+				     */
+	uint32_t user_val_2;        /*!< 30 bits user value available directly in IPC header
+				     * for some notifications
+				     */
+	uint32_t max_payload_size;  /*!< Data size of payload (NotificationCreate updates this
+				     * value to max possible payload size)
+				     */
+	uint8_t *payload;           /*!< Pointer on the payload */
+} NotificationParams;
+
+/*! \brief Defines parameters used by ADSP system during Module Event notification creation. */
+typedef struct _ModuleEventNotification {
+	uint32_t module_instance_id; /*!< Module ID (MS word) + Module Instance ID (LS word) */
+	uint32_t event_id;           /*!< Module specific event ID. */
+	uint32_t event_data_size;    /*!< Size of event_data array in bytes. May be set to 0
+				      * in case there is no data.
+				      */
+	uint32_t event_data[0];      /*< Optional event data (size set to 0 as it is optional
+				      * data)
+				      */
+} ModuleEventNotification;
+
+/*! \brief Defines notification targets supported by ADSP system */
+typedef enum _NotificationTarget {
+	DSP_TO_HOST = 1,  /*!< Notification target is HOST */
+	DSP_TO_ISH = 2    /*!< Notification target is ISH */
+} NotificationTarget;
+
+/*! \brief Defines size of notification header used by ADSP system */
+#define OUTMESSAGE_HEADER_SIZE              28
+#define NOTIFICATION_HEADER_SIZE            48
+#define IXC_MAX_CLIENT_PAYLOAD_SIZE         256
```

Comment:
These 3 look like IPC ABI ?

Reply:
THis is not IPC ABI. This is maximal size of notification calculated for IADK modules. Removed since no needed on adapter side.